app/scripts/components/common/blocks/scrollytelling/index.tsx

```diff
@@ -134,103 +123,84 @@ function getChapterLayerKey(ch: ScrollyChapter) {
 
 /**
  *
- * @param {array} chList List of chapters
+ * @param {array} chList List of chapters with related layers.
  */
-function useMapLayersFromChapters(chList: ScrollyChapter[]) {
+function useMapLayersFromChapters(chList: ScrollyChapter[]): [
+  ResolvedLayer[],
+  string[]
+] {
   // The layers are unique based on the dataset, layer id and datetime.
-  // Filter out scrollytelling block that doesn't have layer first.
+  // First we filter out any scrollytelling block that doesn't have layer.
   const uniqueChapterLayers = useMemo(() => {
     const unique = chList
       .filter(({ showBaseMap }) => !showBaseMap)
-      .reduce(
-        (acc, ch) => acc.set(getChapterLayerKey(ch), ch),
-        new Map<string, ScrollyChapter>()
-      );
-    return Array.from(unique.values());
+      .reduce<Record<string, ScrollyChapter>>((acc, ch) => {
+        const key = getChapterLayerKey(ch);
+        acc[key] = ch;
+        return acc;
+      }, {});
+
+    return Object.values(unique);
+
   }, [chList]);
 
-  // Create an array of datasetId & layerId to pass useAsyncLayers so that the
-  // layers can be loaded. The skipCompare prevents the compare layer to be
-  // loaded, since it will never be used.
+  // Create an array of datasetId & layerId pairs which we can easily validate when creating
+  // the layers array.
   const uniqueLayerRefs = useMemo(() => {
     return uniqueChapterLayers.map(({ datasetId, layerId }) => ({
       datasetId,
       layerId,
-      skipCompare: true
     }));
   }, [uniqueChapterLayers]);
 
-  const asyncLayers = useAsyncLayers(uniqueLayerRefs);
+  // Validate that all layers are defined in the configuration.
+  // They must be defined in the configuration otherwise it is not possible to load them.
+  const reconciledVizDatasets = uniqueLayerRefs.map(({ datasetId, layerId }) => {
+    const layers = datasets[datasetId]?.data.layers;
 
-  // Create a ref to cache each of the async layers.
-  // After the async layer data is loaded from STAC, the layer functions have
-  // to be resolved by the `resolveConfigFunctions`. This function will return a
-  // new object every time causing useEffects that depend on this data to fire
-  // multiple times, even though the data didn't actually change. An example of
-  // this is the `sourceParams` in `MapLayerRasterTimeseries`.
-  // Since the these values only have to be computed once, when the layer loads,
-  // we can use this cache. On every hook run the asyncLayers.map below will
-  // return the cached value if it exists or compute and cache.
```

Comment:
@hanbyul-here @sandrahoang686 I think this cache might not be needed with the new Layer and Map component. Any issues that you see by removing this?

Reply:
I am not 100% confident, but I think it is already being handled at the layer level at this point? https://github.com/NASA-IMPACT/veda-ui/blob/main/app/scripts/components/exploration/components/map/layer.tsx#L53

Reply:
Thought so as well, and there don't seem to be side effects by not using it for the new layer. I guess we can proceed like this.